src/main/java/org/kaazing/netx/ws/internal/DefaultWebSocketFactory.java

```diff
@@ -17,35 +17,37 @@
 package org.kaazing.netx.ws.internal;
 
 import static java.util.Collections.unmodifiableCollection;
+import static java.util.Collections.unmodifiableList;
 import static java.util.Collections.unmodifiableMap;
 
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.ServiceLoader;
 
 import org.kaazing.netx.http.HttpRedirectPolicy;
 import org.kaazing.netx.http.auth.ChallengeHandler;
-import org.kaazing.netx.ws.internal.WebSocketExtension.Parameter;
+import org.kaazing.netx.ws.WebSocket;
+import org.kaazing.netx.ws.WebSocketExtension;
+import org.kaazing.netx.ws.WebSocketFactory;
 import org.kaazing.netx.ws.internal.ext.WebSocketExtensionFactorySpi;
-import org.kaazing.netx.ws.internal.ext.WebSocketExtensionParameterValues;
 
 public final class DefaultWebSocketFactory extends WebSocketFactory {
-    private static final Map<String, WebSocketExtensionFactorySpi>  _extensionFactories;
+    private static final Map<String, WebSocketExtensionFactorySpi>  extensionFactories;
```

Comment:
I see. So, you mean add a method with `@Resource` annotation to inject extension-factories into DefaultWebSocketFactory from WebSocketFactory?